File: 剑指offer/021-栈的压入、弹出序列/myCode.py
from leetcode_tester import Tester

class Solution:
	def IsPopOrder(self, pushV, popV):
		stack = []
		while popV:
			toPop = popV.pop(0)
			while pushV and not pushV[0]==toPop:
				toPush = pushV.pop(0)
				stack.append(toPush)
			if pushV and pushV[0] == toPop:
				toPush = pushV.pop(0)
				stack.append(toPush)
			if not toPop==stack[-1]:
				return False
			stack.pop()
		if not stack:
			return True
			
solution = Solution()
tester = Tester(solution.IsPopOrder)
tester.addTest([1,2,3,4,5],[4,5,3,2,1],True)
tester.addTest([1,2,3,4,5],[4,3,5,1,2],False)
tester.doTest()
		

File: 剑指offer/021-栈的压入、弹出序列/myCode.ts
function IsPopOrder(pushV:Array<number>, popV:Array<number>):boolean{
    let temp = [];
    while(pushV.length){
        let toPush = pushV.shift();
        temp.push(toPush);
        while(popV.length && temp[temp.length-1] === popV[0]){
            temp.pop();
            popV.shift();
        }
    }
    console.log(temp)
    return temp.length === 0;
}

console.log(IsPopOrder([1,2,3,4,5], [4,5,3,2,1]));